cmd/antrea-agent/options.go

```diff
@@ -397,22 +403,23 @@ func (o *Options) setK8sNodeDefaultOptions() {
 	if o.config.HostProcPathPrefix == "" {
 		o.config.HostProcPathPrefix = defaultHostProcPathPrefix
 	}
-	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
-		if o.config.AntreaProxy.ProxyLoadBalancerIPs == nil {
-			o.config.AntreaProxy.ProxyLoadBalancerIPs = new(bool)
-			*o.config.AntreaProxy.ProxyLoadBalancerIPs = true
-		}
-	} else {
-		if o.config.ServiceCIDR == "" {
-			o.config.ServiceCIDR = defaultServiceCIDR
-		}
+	if o.config.AntreaProxy.Enable == nil {
+		o.config.AntreaProxy.Enable = pointer.Bool(true)
+	}
```

Comment:
Do we need these 3 lines in `setExternalNodeDefaultOptions` as well?
It may be needed to avoid a crash if someone uses an old version of the config. I think we did the same in the past with `enablePrometheusMetrics`.

Reply:
It was because `EnablePrometheusMetrics` was accessed directly in `run`. In AntreaProxy case, we have `enableAntreaProxy` and use it in `run`, so it should be fine to leave it nil in ExternalNode case.

Reply:
And we don't need to default it to true?

Reply:
It won't be read in ExternalNode case as `validateK8sNodeOptions` won't even be executed in that case. This implies a question: whether defaulting should be done in a modeless way or on demand. I think currently it's the latter. However, I also wondered if the former is better as it could ensure every config gets a value, either from user config or the default one.